pkg/cmd/create/workflow.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright © 2021 The LitmusChaos Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package create
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gorhill/cronexpr"
+	"github.com/litmuschaos/litmusctl/pkg/apis"
+	"github.com/litmuschaos/litmusctl/pkg/utils"
+
+	"github.com/litmuschaos/litmus/litmus-portal/graphql-server/graph/model"
+
+	"github.com/spf13/cobra"
+)
+
+// workflowCmd represents the project command
+var workflowCmd = &cobra.Command{
+	Use: "workflow",
+	Short: `Create a Chaos Workflow
+	Example:
+	#create a chaos workflow
+	litmusctl create workflow -f workflow.yaml --project-id="d861b650-1549-4574-b2ba-ab754058dd04" --agent-id="1c9c5801-8789-4ac9-bf5f-32649b707a5c"
+
+	Note: The default location of the config file is $HOME/.litmusconfig, and can be overridden by a --config flag
+	`,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		// Fetch user credentials
+		credentials, err := utils.GetCredentials(cmd)
+		utils.PrintError(err)
+
+		var chaosWorkFlowRequest model.ChaosWorkFlowRequest
+
+		workflowManifest, err := cmd.Flags().GetString("file")
+		utils.PrintError(err)
+
+		chaosWorkFlowRequest.ProjectID, err = cmd.Flags().GetString("project-id")
+		utils.PrintError(err)
+
+		// Handle blank input for project ID
+		if chaosWorkFlowRequest.ProjectID == "" {
+			utils.White_B.Print("\nEnter the Project ID: ")
+			fmt.Scanln(&chaosWorkFlowRequest.ProjectID)
+
+			if chaosWorkFlowRequest.ProjectID == "" {
+				utils.Red.Println("⛔ Project ID can't be empty!!")
+				os.Exit(1)
+			}
+		}
+
+		chaosWorkFlowRequest.ClusterID, err = cmd.Flags().GetString("agent-id")
+		utils.PrintError(err)
+
+		// Handle blank input for agent ID
+		if chaosWorkFlowRequest.ClusterID == "" {
+			utils.White_B.Print("\nEnter the Agent ID: ")
+			fmt.Scanln(&chaosWorkFlowRequest.ClusterID)
+
+			if chaosWorkFlowRequest.ClusterID == "" {
+				utils.Red.Println("⛔ Agent ID can't be empty!!")
+				os.Exit(1)
+			}
+		}
+
+		// Perform authorization
```

Comment:
@SarthakJain26 @Saranya-jena Are we not doing these authorization checks in backend while scheduling workflows in gql-server using GRPC?

Reply:
Yes authorization is there in the backend, is this for displaying proper error messages in the CLI @PrayagS ?

Reply:
Yes, this was being done for error messages.

I turned off this code and GQL returns a `permission_denied` error. If it sounds good, I can remove this block and show the same error message after getting the response from the GQL server.